borsh/src/de/mod.rs

```diff
@@ -91,15 +92,17 @@ impl BorshDeserialize for u8 {
 
     #[inline]
     #[doc(hidden)]
-    fn copy_from_bytes(buf: &mut &[u8], out: &mut [Self]) -> Result<bool> {
+    fn copy_from_bytes(buf: &mut &[u8], out: &mut [MaybeUninit<Self>]) -> Result<bool> {
         if buf.len() < out.len() {
             return Err(Error::new(
                 ErrorKind::InvalidInput,
                 ERROR_UNEXPECTED_LENGTH_OF_INPUT,
             ));
         }
         let (front, rest) = buf.split_at(out.len());
-        out.copy_from_slice(front);
+        for (o, f) in out.iter_mut().zip(front.iter()) {
+            o.write(*f);
+        }
```

Comment:
Hm, so the reason why we have this "specialization" in the first place is to be able to optimize this loop. The complier might be smart enough to optimize this anyhow, but I'd prefer to be explicit here.